### DISCUSSION TITLE
Slower than expected performance on GPU and jetson

### DISCUSSION THREAD

**Discussion by GOBish** [STVIR/pysot#387] [General] [answered]
Hi,
I am trying to use a Siam tracker on a Jetson Xavier Nx.

I’ve tried the different models in the model zoo.

I am currently only getting 12-14 FPS using a relatively small image size. (SiamMask is slower at 4-5 FPS).
I did confirm that the GPU is running at 70 to 99 percent.

I would appreciate any help!

**Comment by ZhiyuanChen** [comment]
Well, considering you are running on a device with only 384 CUDA units, I would personally argue this performance is quite unbelievable. Try use much smaller backbone network, e.g. darknet may help.

**Comment by canozcivelek** [comment]
Hi, I'm running the demo with decent performance on the Jetson Nano (using the siamrpn_alex_dwxcorr_otb model). I adjusted the code a little bit to start streaming without select ROI on the first frame. So stream starts directly, and when I click on an object, then it starts tracking. My question is that when I first select a ROI, it freezes for around 8-10 seconds before the tracking starts. Any suggestions on how to prevent freezing when the code runs tracker.init() line? Is multiprocessing an option for example? Thanks!

**Comment by ZhiyuanChen** [comment]
> Hi, I'm running the demo with decent performance on the Jetson Nano (using the siamrpn_alex_dwxcorr_otb model). I adjusted the code a little bit to start streaming without select ROI on the first frame. So stream starts directly, and when I click on an object, then it starts tracking. My question is that when I first select a ROI, it freezes for around 8-10 seconds before the tracking starts. Any suggestions on how to prevent freezing when the code runs tracker.init() line? Is multiprocessing an option for example? Thanks!

Again I think it should be expected, the freezeing is the time when network forward the template. And again try use much smaller backbone network, e.g. darknet may help.

**Comment by canozcivelek** [comment, marked as answer]
Hi, thanks for the feedback. I was able to work around the issue by initiate tracking at a given coordinate (hard coded inside the script) in the first frame, then immediately stop tracking process by emptying the bbox. This way, when the camera starts streaming, it would already have done an initial tracking (network forward template) and when selecting a new ROI, tracking would start immediately. An interesting solution no? :)
